src/dsl.jl

```diff
@@ -828,83 +805,123 @@ function read_observed_options(options, species_n_vars_declared, ivs_sorted; req
 
             # Adds the observable to the list of observable names.
             # This is required for filtering away so these are not added to the ReactionSystem's species list.
             # Again, avoid this check if we have interpolated the variable.
             is_escaped_expr(obs_eq.args[2]) || push!(obs_syms.args, obs_name)
         end
 
-        # If nothing was added to `observed_vars`, it has to be modified not to throw an error.
-        (striplines(observed_vars) == striplines(Expr(:block, :(@variables)))) &&
-            (observed_vars = :())
+        # If nothing was added to `obsexpr`, it has to be modified not to throw an error.
+        (striplines(obsexpr) == striplines(Expr(:block, :(@variables)))) &&
+            (obsexpr = :())
     else
-        # If option is not used, return empty expression and vector.
-        observed_vars = :()
-        observed_eqs = :([])
+        # If observables option is not used, return empty expression and vector.
+        obsexpr = :()
+        obs_eqs = :([])
         obs_syms = :([])
     end
-    return observed_vars, observed_eqs, obs_syms
+
+    return obsexpr, obs_eqs, obs_syms
 end
 
-# From the input to the @observables options, creates a vector containing one equation for each observable.
-# Checks separate cases for "@observables O ~ ..." and "@observables begin ... end". Other cases errors.
-function make_observed_eqs(observables_expr)
-    if observables_expr.head == :call
-        return :([$(observables_expr)])
-    elseif observables_expr.head == :block
-        observed_eqs = :([])
-        for arg in observables_expr.args
-            push!(observed_eqs.args, arg)
-        end
-        return observed_eqs
+# From the input to the @observables options, create a vector containing one equation for
+# each observable. `option_block_form` handles if single line declaration of `@observables`,
+# i.e. without a `begin ... end` block, was used.
+function make_obs_eqs(observables_expr)
+    observables_expr = option_block_form(observables_expr)
+    obs_eqs = :([])
+    foreach(arg -> push!(obs_eqs.args, arg), observables_expr.args)
+    return obs_eqs
+end
+
+# Reads the combinatorial ratelaw options, which determines if a combinatorial rate law should
+# be used or not. If not provided, use the default (true).
+function read_combinatoric_ratelaws_option(options)
+    return haskey(options, :combinatoric_ratelaws) ?
+        get_block_option(options[:combinatoric_ratelaws]) : true
+end
+
+# Finds the time independent variable, and any potential spatial independent variables.
+# Returns these (individually and combined), as well as an expression for declaring them.
+function read_ivs_option(options)
+    # Creates the independent variables expressions (depends on whether the `ivs` option was used).
+    if haskey(options, :ivs)
+        ivs = Tuple(extract_syms(options, :ivs))
+        ivsexpr = copy(options[:ivs])
+        ivsexpr.args[1] = Symbol("@", "parameters")
```

Comment:
```suggestion
        ivsexpr.args[1] = Symbol("@", "parameters")
```
We should probably switch to `@independent_variables` as MTK is going to use that in the future (for now it just wraps `@parameters` but that will change).

Reply:
Agreed, changing to `@independent_variables`

Reply:
agreed, I've added this in now